lib/Gitlab/Client.php

```diff
@@ -54,48 +60,50 @@ class Client
     const AUTH_OAUTH_TOKEN = 'oauth_token';
 
     /**
-     * @var array
+     * @var History
      */
-    private $options = array(
-        'user_agent'  => 'php-gitlab-api (http://github.com/m4tthumphrey/php-gitlab-api)',
-        'timeout'     => 60
-    );
-
-    private $baseUrl;
+    private $responseHistory;
 
     /**
-     * The Buzz instance used to communicate with Gitlab
-     *
-     * @var HttpClient
+     * @var Builder
      */
-    private $httpClient;
+    private $httpClientBuilder;
 
     /**
      * Instantiate a new Gitlab client
      *
-     * @param string               $baseUrl
-     * @param null|ClientInterface $httpClient Buzz client
-     * @param array                $options
+     * @param Builder $httpClientBuilder
      */
-    public function __construct($baseUrl, ClientInterface $httpClient = null, array $options = array())
+    public function __construct(Builder $httpClientBuilder = null)
     {
-        foreach ($options as $name => $value) {
-            $this->setOption($name, $value);
-        }
-
-        $httpClient = $httpClient ?: new Curl();
-        $httpClient->setTimeout($this->options['timeout']);
-        $httpClient->setVerifyPeer(false);
+        $this->responseHistory = new History();
+        $this->httpClientBuilder = $httpClientBuilder ?: new Builder();
+
+        $this->httpClientBuilder->addPlugin(new GitlabExceptionThrower());
+        $this->httpClientBuilder->addPlugin(new HistoryPlugin($this->responseHistory));
+        $this->httpClientBuilder->addPlugin(new RedirectPlugin());
+        $this->httpClientBuilder->addPlugin(new ApiVersion());
+
+        $userAgent = isset($options['user_agent'])
+            ? $options['user_agent']
+            : 'php-gitlab-api (http://github.com/m4tthumphrey/php-gitlab-api)'
+        ;
+        $this->setUserAgent($userAgent);
+        $this->setUrl('https://gitlab.com');
+    }
 
-        $this->baseUrl     = $baseUrl;
-        $this->httpClient  = new HttpClient($this->baseUrl, $this->options, $httpClient);
+    /**
+     * Create a Gitlab\Client using a HttpClient.
+     *
+     * @param HttpClient $httpClient
+     *
+     * @return Client
+     */
+    public static function createWithHttpClient(HttpClient $httpClient)
```

Comment:
Hm, I know I asked before. Maybe create a factory function for URL?

`Client::create($url)`?